llm_sandbox/core/mixins.py

```
@@ -44,67 +44,87 @@ def copy_from_container(self, container: Any, src: str) -> tuple[bytes, dict]:
-            msg = f"Operation timed out after {timeout} seconds"
-            raise SandboxTimeoutError(msg)
+        try:
+            # Wait for completion or timeout
+            if not completed.wait(timeout):
+                msg = f"Operation timed out after {timeout} seconds"
+
+                # Optional: Force container-level kill for true cancellation
+                handler = getattr(self, "_handle_timeout", None)
+                if force_kill_on_timeout and callable(handler):
+                    try:
+                        handler()  # pyright: ignore[reportAttributeAccess]
```

Pylint (3.3.7)

[error] handler is not callable

(E1102)